Makefile

```diff
@@ -6,7 +6,21 @@ TEST_IMG_NAME ?= wasmtest:latest
 RUNTIMES ?= wasmedge wasmtime wasmer
 CONTAINERD_NAMESPACE ?= default
 
-TARGET ?= 
+ifeq ($(CARGO),cross)
+# Set the default target as defined in Cross.toml
+TARGET ?= $(shell uname -m)-unknown-linux-musl
+# When using `cross` we need to run the tests outside the `cross` container.
+# We stop `cargo test` from running the tests with the `--no-run` flag.
+# We then need to run the generate test binary manually.
+# For that we use `--message-format=json` and `jq` to find the name of the binary, `xargs` and execute it.
+TEST_ARGS_SEP= --no-run --color=always --message-format=json | \
+	jq -R '. as $$line | try (fromjson | .executable | strings) catch ($$line+"\n" | stderr | empty)' -r | \
+	xargs -I_ sudo -E ./_
+else
+TARGET ?= $(shell rustc --version -v | sed -En 's/host: (.*)/\1/p')
```

Comment:
in the case of #356 if I am running ubuntu locally then run `make test/k8s-wasmtime` it will fail.

Reply:
Ah, I see, I think that's correct.
I think we could force using `cross-rs` here.

Reply:
The latest push forces using `cross-rs` for k8s testing.